runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PipelineResources.java

```diff
@@ -61,7 +61,10 @@
     List<String> files = new ArrayList<>();
     for (URL url : ((URLClassLoader) classLoader).getURLs()) {
       try {
-        files.add(new File(url.toURI()).getAbsolutePath());
+        File file = new File(url.toURI());
+        if (file.exists()) {
```

Comment:
Since SparkContext will reject all directories [1], it is probably best to exclude directories too.

```suggestion
        if (file.exists() && !file.isDirectory()) {
```

[1] https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L1796-L1802

Reply:
Actually we cannot exclude directories. Excluding directories will cause flink runner related unit test case fail.